gcloud/start-remote-job

```diff
@@ -86,60 +90,71 @@ safety_dir = os.path.abspath(os.path.join(__file__, '../../'))
 ssh_cmd = os.path.abspath(os.path.join(__file__, '../ssh'))
 
 print("Cloning temporary repo...")
-result = sp.run(
-    f"git clone --no-local --single-branch --branch {git_tag} . ./tmp-repo",
-    shell=True, **quiet)
-if not args.tag:
-    print("Removing temporary tag...")
-    sp.run(f"git tag -d {git_tag}", shell=True, **quiet)
-if result.returncode != 0:
-    print("Error cloning the repo.")
-    exit(1)
+
+try:
+    tmp_repo = tempfile.TemporaryDirectory(prefix="safelife-job-")
+    result = sp.run(
+        f"git clone --no-local --single-branch --branch {git_tag} . {tmp_repo.name}",
+        shell=True, **quiet)
+    if result.returncode != 0:
+        print("Error cloning the repo.")
+        exit(1)
 
 # Also copy over some files that are not (necessarily) in source control.
-extra_files = [
-    'run-info.md',
-    'run-notes.txt',
-    'wandb/settings',
-]
-for extra_file in extra_files:
-    src_file = os.path.join(safety_dir, extra_file)
-    dst_file = os.path.join(safety_dir, 'tmp-repo', extra_file)
-    if os.path.exists(src_file):
-        os.makedirs(os.path.dirname(dst_file), exist_ok=True)
-        shutil.copyfile(src_file, dst_file)
-
-
-print("Syncing repo to the cloud...")
-result = sp.run([
-    'rsync', '--rsh', ssh_cmd, '-ra',
-    './tmp-repo/', args.instance_name + ':' + src_dir])
-shutil.rmtree("./tmp-repo")
-if result.returncode != 0:
-    exit(1)
+    extra_files = [
+        'run-info.md',
+        'run-notes.txt',
+        'wandb/settings',
+    ]
+    for extra_file in extra_files:
+        src_file = os.path.join(safety_dir, extra_file)
+        dst_file = os.path.join(tmp_repo.name, extra_file)
+        if os.path.exists(src_file):
+            os.makedirs(os.path.dirname(dst_file), exist_ok=True)
+            shutil.copyfile(src_file, dst_file)
+
+    print("Syncing repo to the cloud...")
+    result = sp.run([
+        'rsync', '--rsh', ssh_cmd, '-ra',
+        tmp_repo.name + '/', args.instance_name + ':' + src_dir])
```

Comment:
@pde Note that I've added a slash after `tmp_repo.name`. Without this it was getting uploaded to the wrong directory. Double check that it's not doing something different on your system.